### PR TITLE
test: Add e2e for data transformation expressions

### PR DESCRIPTION
**Pull request by ivov**
https://linear.app/n8n/issue/N8N-6089/e2e-happy-path-for-workflow-building

### CODE REVIEW

File: cypress/pages/workflow.ts [hunk 1]
```diff
@@ -86,11 +86,14 @@ export class WorkflowPage extends BasePage {
...
-		addInitialNodeToCanvas: (nodeDisplayName: string) => {
+		addInitialNodeToCanvas: (
+			nodeDisplayName: string,
+			{ preventNdvClose } = { preventNdvClose: true },
+		) => {
 			this.getters.canvasPlusButton().click();
 			this.getters.nodeCreatorSearchBar().type(nodeDisplayName);
 			this.getters.nodeCreatorSearchBar().type('{enter}');
-			cy.get('body').type('{esc}');
+			if (!preventNdvClose) cy.get('body').type('{esc}');
```
[the remark this answers is not in the document]
Comment (ivov) [reply cypress/pages/workflow.ts]: Renamed. Did it to stay consistent with next helper which also uses `preventNdvClose`.

File: cypress/e2e/14-data-transformation-expressions.cy.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+//             utils
+// ----------------------------------
+
+const addWithPinData = (nodeTypeName: string, pinData: object[]) => {
+	wf.actions.addInitialNodeToCanvas(nodeTypeName, { preventNdvClose: true });
+	ndv.actions.setPinnedData(pinData);
+	ndv.actions.close();
+};
+
+const addSet = () => {
```
[the remark this answers is not in the document]
Comment (ivov) [reply cypress/e2e/14-data-transformation-expressions.cy.ts]: Removed helper and used above. I was on the fence whether to make it reusable.

[the remark this answers is not in the document]
Comment (mutdmour) [reply cypress/e2e/14-data-transformation-expressions.cy.ts]: it's your call. was wondering.. could def see first function being reused though

**Review by mutdmour — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.215.1`

**Pull request merged**